### ISSUE TITLE
Rule for certain hanging indents

### ISSUE THREAD

**Issue by AckslD** [astral-sh/ruff#21596]
### Summary

It would be great to have the ability to enable rules which dissallow indentation based on for example a function name instead of eg 4 spaces. Ie to disallow:

```python
def my_really_long_function_name(a, 
                                 b)
```
but prefer
```python
def my_really_long_function_name(
    a, 
    b,
)
```

Something like https://github.com/deniskrumko/flake8-hangover.

I wanted to first check if there could be an interest for this. If so I can try to make it a bit more precise and also happy to try to work on the actual implementation once agreed upon.

**Comment by MichaReiser** [comment]
Hi @AckslD 

I'll merge this into https://github.com/astral-sh/ruff/issues/9216 as adding an option outlined in https://github.com/astral-sh/ruff/issues/9216#issuecomment-2904749624 would do exactly this

**Comment by AckslD** [comment]
Great, thanks @MichaReiser !

**Comment by adamsol** [comment]
While this issue has been marked as a duplicate of #9216, they don't actually seem to be related. `COM812` won't apply in situations like the first OP's example.

Just to recap, [PEP 8](https://peps.python.org/pep-0008/#indentation) describes two possible indentation styles for multiline expressions: visual indent (the first example) and hanging indent (the second example).

The idea is a linter rule that allows only one of the styles for consistency - the hanging indent, as it is more versatile.

This style is generally consistent with Ruff's autoformatter, but I only want to enforce this specific formatting with a linter, while leaving the rest of the code formatted manually.

I've experimented with Ruff and managed to implement the corresponding rule from `flake8-hangover`, as the logic is very simple: #23809. It turns out the rule may be useful even alongside the formatter, as noted in the PR.

**Comment by MichaReiser** [comment]
@adamsol this is marked as a duplicate of #9216 because the discussed solution is to add an option to `COM812` that would make it enforce your desired style by default, but allow opt-out for users that prefer the formatter-incompatible style.

**Comment by adamsol** [comment]
The thing is that it *won't* enforce this style. Could you please check yourself that `COM812` doesn't complain on visual indentation? You already noticed something similar in https://github.com/astral-sh/ruff/issues/9216#issuecomment-1961688768.

**Comment by MichaReiser** [comment]
Right. I still don't think we should introduce a new rule that overlaps with COM812. Instead, we should consider introducing a new option for COM812 to capture a narrower scope (and be compatible with the formatter). Ideally, we want to have as few options as possible.

**Comment by adamsol** [comment]
How exactly does `HG005` overlap with `COM812`? One is about bracket alignment, the other is about trailing commas. Users may want either of them, both, or none.

I'm pretty sure these are completely separate issues, and removing the duplicate label would help reduce further confusion. Or if you think they're actually duplicates, wouldn't it mean that my PR solves #9216?

**Comment by MichaReiser** [comment]
Let's open a new issue and discuss it there. My main concern is formatter compatibility (e.g. hugging parens). We've also haven't added any new formatter lints in a while, so while the rule might be different, I'm not sure it's something we want to add right now.
